### ISSUE TITLE
Couldn't get it to work when devtool is set to "cheap-source-map"

### ISSUE THREAD

**Issue by rosenfeld** [mzgoddard/hard-source-webpack-plugin#252]
It works with "source-map" though, but it takes longer to run for the first time, as expected. If you could fix it to work with cheap-source-map that would be great, but otherwise it would be interesting to at least document which kind of source maps are currently supported by the plugin.

Thanks for this plugin by the way :)

**Comment by mzgoddard** [comment]
Hey @rosenfeld, thanks for filing this issue. Can you share any info on how `cheap-source-map` isn't working with `hard-source` for you? Does it not finish building? Does webpack suddenly quit back to your shell? Does it build but the source maps don't work?

It would help me if you can post whatever parts you can of your webpack config and dependencies list.

Also it'd be great if you could try hard-source's `0.6.0-alpha` version (latest is `0.5`). It does a lot things differently that should be less prone to these kinds of errors.

**Comment by rosenfeld** [comment]
Ok, I just got late at home, so I won't have much time now to provide you with that information. I'll see if I can reproduce it with a small project tomorrow, but for now, here's what I see in the second run with HardSourceWebpackPlugin when devtool is set to cheap-source-map:

```
Unhandled rejection TypeError: Cannot read property 'sources' of undefined

at fromStringWithSourceMap (node_modules/source-list-map/lib/fromStringWithSourceMap.js:11:20)
at HardSource.listMap (node_modules/hard-source-webpack-plugin/lib/hard-source.js:140:34)                         
    at ConcatSource.listMap (node_modules/webpack-sources/lib/ConcatSource.js:72:18)                                  
    at ConcatSource.proto.sourceAndMap (/node_modules/webpack-sources/lib/SourceAndMapMixin.js:25:16)                  
    at CachedSource.sourceAndMap (node_modules/webpack-sources/lib/CachedSource.js:56:31)                             
    at getTaskForFile (node_modules/webpack/lib/SourceMapDevToolPlugin.js:33:30)                                      
    at chunk.files.forEach.file (node_modules/webpack/lib/SourceMapDevToolPlugin.js:91:21)                            
    at Array.forEach (<anonymous>)                                                  
    at node_modules/webpack/lib/SourceMapDevToolPlugin.js:89:18                                                       
    at Array.forEach (<anonymous>)                                                  
    at Compilation.<anonymous> (node_modules/webpack/lib/SourceMapDevToolPlugin.js:88:12)
```

**Comment by rosenfeld** [comment]
Nevermind, this is fixed in version 0.6.0-alpha.2. So, I'm closing this issue, but please let me know once the next stable version is released.

**Comment by rosenfeld** [comment]
Anyway, for production builds I can't use cheap-source-map anyway because the Uglifyjs plugin doesn't support it. Thanks to hard-source plugin this is not a big deal for most deploys ;)
